src/Symfony/Component/Validator/ConstraintValidator.php

```diff
@@ -93,6 +93,10 @@ protected function formatValue(mixed $value, int $format = 0): string
                 return $value->__toString();
             }
 
+            if ($value instanceof \UnitEnum) {
+                return $value->name;
```

Comment:
I'd suggest going with `$value::class.'::'.$value->name`.
can you please add a test case about it?

Reply:
I don't see why it is better to check it with your code `$value::class.'::'.$value->name`. I think it is more readable with `instanceof`. Is there a case in which it can fail?

Reply:
I'm not commenting on the check but on the return value.
We should return the name of the class + the name of case, not only the former.
`$value::class.'::'.$value->name`

Reply:
ohhh, I see it now. I will change it.

Reply:
Mmm, with that change it's a bit too verbose as it returns the full qualified class name. Although as this is more of a developer thing than something which would be seen by clients it could be ok.